File: its-my-training-day/leetcode-top-75/questions/trees/tree-dfs/max-depth-binary-tree/solution.ts
import {
  ArrayList,
  Stack,
  Queue,
  ListNode,
  TreeNode,
  MinHeap,
  MaxHeap,
} from "../../../../shared/typescript/ds";

export const maxDepthBinaryTree = (root: TreeNode | null): number => {
  if (root === null) return 0;
  const left = maxDepthBinaryTree(root.left);
  const right = maxDepthBinaryTree(root.right);
  return 1 + (left > right ? left : right);
};

File: its-my-training-day/leetcode-top-75/questions/trees/tree-dfs/max-depth-binary-tree/solution.py
import os
import sys
from typing import List, Dict, Optional, Tuple

ROOT_DIR = os.path.abspath(os.path.join(os.path.dirname(__file__), "../../../../"))
if ROOT_DIR not in sys.path:
    sys.path.append(ROOT_DIR)

from shared.python.ds import ArrayList, ListNode, Stack, Queue, TreeNode, MinHeap, MaxHeap

def max_depth_binary_tree(root: Optional[TreeNode]) -> int:
    if root is None:
        return 0
    left = max_depth_binary_tree(root.left)
    right = max_depth_binary_tree(root.right)
    return 1 + (left if left > right else right)
